File: competition/04-sort/sort.hs
-- Sort a pseudo-random integer list with the stdlib stable sort, then fold an
-- order-sensitive checksum over the result.
import Data.List (foldl', sort)

main :: IO ()
main = do
  let n = 100000 :: Int
      xs = [(i * 2654435761) `mod` 100003 | i <- [0 .. n - 1]]
      sorted = sort xs
      checksum = foldl' (\acc v -> (acc * 31 + v) `mod` 1000000007) 0 sorted
  print n
  print checksum

File: competition/04-sort/sort.lua
-- Sort a pseudo-random integer list, then fold an order-sensitive checksum.
local n = 100000
local xs = {}
for i = 0, n - 1 do xs[i + 1] = (i * 2654435761) % 100003 end
table.sort(xs)
local checksum = 0
for i = 1, n do checksum = (checksum * 31 + xs[i]) % 1000000007 end
print(n)
print(checksum)
